apps/astarte_appengine_api/lib/astarte_appengine_api_web/plug/group_name_decoder.ex

```diff
@@ -0,0 +1,13 @@
+defmodule Astarte.AppEngine.APIWeb.Plug.GroupNameDecoder do
+  @moduledoc """
+  This plug decodes a group name, which may have been coded
+  to remove the forward slash
+  """
+  def init(default), do: default
+
+  def call(%Plug.Conn{path_params: %{"group_name" => group_name}} = conn, _) do
+    %Plug.Conn{conn | path_params: %{conn.path_params | "group_name" => URI.decode(group_name)}}
+  end
```

Comment:
This feels a bit obscure, what about:
```suggestion
  def call(%Plug.Conn{path_params: path_params} = conn, _) do
    %{"group_name" => encoded_group_name} = path_params
    decoded_group_name = URI.decode(encoded_group_name)
    normalized_path_params = %{path_params | "group_name" => decoded_group_name}
    %Plug.Conn{conn | path_params: normalized_path_params}
  end
```

Reply:
...no please...

Reply:
Otherwise, you can use [`Map.replace/3`](https://hexdocs.pm/elixir/1.12/Map.html#replace/3) to remove a level of map update